ntbs-service/DataMigration/NotificationMapper.cs

```diff
@@ -221,22 +224,34 @@ private async Task<IEnumerable<IList<Notification>>> GetGroupedResultsAsNotifica
             IEnumerable<TreatmentEvent> notificationOutcomeEvents)
         {
             return notification.ClinicalDetails.IsPostMortem == true
-                ? TreatmentEventsForPostMortemNotification(rawNotification)
+                ? TreatmentEventsForPostMortemNotification(rawNotification, notificationOutcomeEvents)
                 : TreatmentEventsForPreMortemNotification(
                     notification,
                     rawNotification,
                     notificationTransferEvents,
                     notificationOutcomeEvents);
         }
 
-        private List<TreatmentEvent> TreatmentEventsForPostMortemNotification(MigrationDbNotification rawNotification)
+        private List<TreatmentEvent> TreatmentEventsForPostMortemNotification(MigrationDbNotification rawNotification,
+            IEnumerable<TreatmentEvent> notificationOutcomeEvents)
         {
             // For post mortem cases the death event is the ONLY event we want to import so the final outcome is
             // correctly reported.
-            return new List<TreatmentEvent>
+            var deathEvents = notificationOutcomeEvents.Where(e => _diedOutcomeIds.Contains(e.TreatmentOutcomeId ?? 0)).ToList();
+            TreatmentEvent eventToReturn;
+            if (deathEvents.Any())
+            {
+                eventToReturn = deathEvents.OrderByDescending(e => e.EventDate).First();
```

Comment:
If there are multiple death events (is that really something that can happen?) then do we care which one we take? I would prefer to just take `.First()` here if we can.

Reply:
I don't think this is very likely to happen, but calling `.Single()` seemed a little risky. I can change this to just `.First()`